### PR TITLE
Support more compiler versions

### PR DESCRIPTION
**Pull request by mato**
Support OCaml up to `4.04.*+*`.

Also reinstate -fno-strict-aliasing and -fwrapv to OCAML_CFLAGS.

### CODE REVIEW

File: configure.sh [hunk 1]
```diff
@@ -21,9 +21,12 @@ fi
 FREESTANDING_CFLAGS="$(pkg-config --cflags ${PKG_CONFIG_DEPS})"
 
 case $(ocamlopt -version) in
-    4.02.3)
+    4.02.[3-9])
```
Comment (hannesm) [inline configure.sh]: I think 4.02.3 will be the last release of the 4.02 series, no need for the complexity here

File: configure.sh [hunk 2]
```diff
@@ -21,9 +21,12 @@ fi
...
         ;;
-    4.03.0)
+    4.03.[0-9])
```
Comment (hannesm) [inline configure.sh]: same here, there won't be a 4.03.X with X != 0.

**Comment by mato** [comment]
Waiting to merge this until I can do an end-to-end test with mirage on 4.04.

**Comment by mato** [comment]
This will have to wait until at least janestreet/ppx_optcomp#2 lands, so leaving unmerged for now.

**Comment by hannesm** [comment]
@mato which package on Mirage side depends on `ppx_optcomp`?

**Comment by mato** [comment]
@hannesm Just been through this with @yomimono, `ipaddr` is the culprit.

**Comment by mato** [comment]
Rebased and re-tested against 4.04.0 release, MirageOS HTTP unikernel running on top happily serves 100k+ requests, so merging.

**Pull request merged**